backend/sirius-web-emf/src/main/java/org/eclipse/sirius/web/emf/services/EMFValidationService.java

```diff
@@ -77,7 +108,29 @@ public EMFValidationService(EValidator.Registry eValidatorRegistry) {
         // @formatter:on
     }
 
-    private org.eclipse.sirius.web.validation.Diagnostic convertToDiagnostic(Diagnostic diagnostic) {
+    private org.eclipse.sirius.web.validation.Diagnostic convertToValidationDiagnostic(Diagnostic diagnostic) {
+        String kind = this.getKind(diagnostic);
+
+        // @formatter:off
+        return org.eclipse.sirius.web.validation.Diagnostic.newDiagnostic(UUID.randomUUID())
+                .kind(kind)
+                .message(diagnostic.getMessage())
+                .build();
+        // @formatter:on
+    }
+
+    private org.eclipse.sirius.web.forms.validation.Diagnostic convertToFormDiagnostic(Diagnostic diagnostic) {
+        String kind = this.getKind(diagnostic);
+
+     // @formatter:off
+        return org.eclipse.sirius.web.forms.validation.Diagnostic.newDiagnostic(UUID.randomUUID())
+                .kind(kind)
+                .message(diagnostic.getMessage())
+                .build();
+        // @formatter:on
+    }
```

Comment:
I missed this part before, but why do you need to convert the EMF diagnostics to those POJOs? You do not need to do that here, unless I missed something. The whole point of the API that we have defined in the ADR was to support any diagnostic POJOs.

Reply:
Done